dali/python/nvidia/dali/pipeline.py

```diff
@@ -990,3 +996,63 @@ def iter_setup(self):
         For example, one can use this function to feed the input
         data from NumPy arrays."""
         pass
+
+
+def _discriminate_args(**kwargs):
+    """Split args on those applicable to Pipeline constructor and the rest."""
+    fca = inspect.getfullargspec(Pipeline.__init__)  # Full Ctor Args
+    ctor_args = dict(filter(lambda x: x[0] in fca.args or x[0] in fca.kwonlyargs, kwargs.items()))
+    fn_args = dict(filter(lambda x: x[0] not in ctor_args.keys(), kwargs.items()))
+    return ctor_args, fn_args
+
+
+def pipeline(fn=None, **pipeline_kwargs):
+    """
+    Decorator for wrapping functions that define DALI pipelines.
+
+    **Usage**
+
+    First, implement a function, that defines a DALI pipeline.
+    Such function shall return DALI's DataNodes. These return
```

Comment:
It can return not only DataNodes.

I'm missing the information, that this decorator will turn this function into something like Pipeline factory. And that the returns of this function become output of the Pieline (for example .run()) and the function will return Pipeline instances.

Reply:
Done